ompi/mpi/fortran/use-mpi-tkr/mpi.F90

```diff
@@ -51,6 +52,10 @@ module mpi
 
   include "mpi-f90-interfaces.h"
   include "pmpi-f90-interfaces.h"
+#if OMPI_FORTRAN_HAVE_BIND_C_TYPE
```

Comment:
I think you need to add `OMPI_FORTRAN_HAVE_BIND_C_TYPE` to `ompi/mpi/fortran/configure-fortran-output.h.in`.

Reply:
this is correct.

now I am facing an other issue, and it seems the only option left is to define `type(MPI_Status)` in its own module ...

I will implement this.